packages/adapter-components/src/client/http_client.ts

```diff
@@ -256,7 +256,7 @@ export abstract class AdapterHTTPClient<
       }
       if (e.response !== undefined) {
         logResponse(e.response)
-        throw new HTTPError(`Failed to ${method} ${url} with error: ${e}`, _.pick(e.response, ['status', 'data']))
+        throw new HTTPError(`Failed to ${method} ${url} with error: ${e}`, { ..._.pick(e.response, ['status', 'data']), headers: this.extractHeaders(e.response.headers) })
```

Comment:
nit: not sure it makes sense to use `_.pick()` anymore - 
```suggestion
        throw new HTTPError(`Failed to ${method} ${url} with error: ${e}`, {
          status: e.response.status,
          data: e.response.data,
          headers: this.extractHeaders(e.response.headers),
        })
```